src/screens/connect-extension/connect-extension.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { View, Image } from 'react-native';
+import { Text, Button } from '../../library';
+import { withTheme, IThemeProps } from '../../core/theme/with-theme';
+import { translate } from '../../core/i18n';
+import stylesProvider from './styles';
+import { smartConnect } from '../../core/utils/smart-connect';
+import { WalletConnectClient } from '../../core/wallet-connect/wallet-connect-client';
+import { QrModalReader } from '../../components/qr-modal/qr-modal';
+import bind from 'bind-decorator';
+import { normalize } from '../../styles/dimensions';
+import { Icon } from '../../components/icon';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { DialogComponent } from '../../components/dialog/dialog-component';
+import { deleteFromStorage } from '../../core/secure/storage';
+import { WC_CONNECTION } from '../../core/constants/app';
+
+export const navigationOptions = () => ({
+    title: translate('ConnectExtension.title')
+});
+
+interface IState {
+    isConnected: boolean;
+}
+
+export class ConnectExtensionScreenComponent extends React.Component<
+    IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    public static navigationOptions = navigationOptions;
+    public qrCodeScanner: any;
+    public connectionInterval: any;
+
+    constructor(props: IThemeProps<ReturnType<typeof stylesProvider>>) {
+        super(props);
+        this.state = {
+            isConnected: WalletConnectClient.isConnected()
+        };
+    }
+
+    public componentDidMount() {
+        this.connectionInterval = setInterval(() => {
```

Comment:
this is weird, basically we verify every 500ms if we have connection, and if we don't we delete connection details.
There may be cases when the connection could take longer

Reply:
i think we should not delete connection

Reply:
also maybe Wallet connect has some events for connect and disconnect events, this way we could refactor this interval

Reply:
I made some changes and improvements, but I left a TODO with one more issue to solve. I'll continue to search for a fix.